dipy/viz/actor.py

```diff
@@ -0,0 +1,648 @@
+                is_colormap = True
+
+    vtk_colors.SetName("Colors")
+
+    # Create the poly_data
+    poly_data = vtk.vtkPolyData()
+    poly_data.SetPoints(vtk_points)
+    poly_data.SetLines(vtk_lines)
+    poly_data.GetPointData().SetScalars(vtk_colors)
+    return poly_data, is_colormap
+
+
+def colormap_lookup_table(scale_range=(0, 1), hue_range=(0.8, 0),
+                          saturation_range=(1, 1), value_range=(0.8, 0.8)):
+    """ Lookup table for the colormap
+
+    Parameters
+    ----------
+    scale_range : tuple
+        It can be anything e.g. (0, 1) or (0, 255). Usually it is the mininum
+        and maximum value of your data.
+    hue_range : tuple of floats
+        HSV values (min 0 and max 1)
+    saturation_range : tuple of floats
+        HSV values (min 0 and max 1)
+    value_range : tuple of floats
+        HSV value (min 0 and max 1)
+
+    Returns
+    -------
+    lookup_table : vtkLookupTable
+
+    """
+    lookup_table = vtk.vtkLookupTable()
+    lookup_table.SetRange(scale_range)
+    lookup_table.SetTableRange(scale_range)
+
+    lookup_table.SetHueRange(hue_range)
+    lookup_table.SetSaturationRange(saturation_range)
+    lookup_table.SetValueRange(value_range)
+
+    lookup_table.Build()
+    return lookup_table
+
+
+def scalar_bar(lookup_table=None, title=" "):
+    """ Default scalar bar actor for a given colormap
+
+    Parameters
+    ----------
+    lookup_table : vtkLookupTable or None
+        If None then ``colormap_lookup_table`` is called with default options.
+    title : str
+
+    Returns
+    -------
+    scalar_bar : vtkScalarBarActor
+
+    """
+    lookup_table_copy = vtk.vtkLookupTable()
+    # Deepcopy the lookup_table because sometimes vtkPolyDataMapper deletes it
+    if lookup_table is None:
+        lookup_table = colormap_lookup_table()
+    lookup_table_copy.DeepCopy(lookup_table)
+    scalar_bar = vtk.vtkScalarBarActor()
+    scalar_bar.SetTitle(title)
+    scalar_bar.SetLookupTable(lookup_table_copy)
+    scalar_bar.SetNumberOfLabels(6)
+
+    return scalar_bar
+
+
+def _arrow(pos=(0, 0, 0), color=(1, 0, 0), scale=(1, 1, 1), opacity=1):
+    ''' Internal function for generating arrow actors.
+    '''
+    arrow = vtk.vtkArrowSource()
+    # arrow.SetTipLength(length)
+
+    arrowm = vtk.vtkPolyDataMapper()
+
+    if major_version <= 5:
+        arrowm.SetInput(arrow.GetOutput())
+    else:
+        arrowm.SetInputConnection(arrow.GetOutputPort())
+
+    arrowa = vtk.vtkActor()
+    arrowa.SetMapper(arrowm)
+
+    arrowa.GetProperty().SetColor(color)
+    arrowa.GetProperty().SetOpacity(opacity)
+    arrowa.SetScale(scale)
+
+    return arrowa
+
+
+def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
+         opacity=1):
+    """ Create an actor with the coordinate's system axes where
+    red = x, green = y, blue = z.
+
+    Parameters
+    ----------
+    scale : tuple (3,)
+        axes size e.g. (100, 100, 100)
```

Comment:
Seems you're right. The numpy docstring standard, which is what we rely on,
seems to be ambiguous:

It says: "They can also be detailed in the description".

Nevertheless, I find it to be useful, especially in cases where default
value is inferred, rather than specified. In this case, you are right that
it appears in the signature anyway.




On Sat, Oct 10, 2015 at 10:27 AM, Eleftherios Garyfallidis <
notifications@github.com> wrote:

> In dipy/viz/actor.py
> <https://github.com/nipy/dipy/pull/587#discussion_r41699339>:
>
> > +    arrowa.GetProperty().SetColor(color)
> > +    arrowa.GetProperty().SetOpacity(opacity)
> > +    arrowa.SetScale(scale)
> > +
> > +    return arrowa
> > +
> > +
> > +def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
> > +         opacity=1):
> > +    """ Create an actor with the coordinate's system axes where
> > +    red = x, green = y, blue = z.
> > +
> > +    Parameters
> > +    ----------
> > +    scale : tuple (3,)
> > +        axes size e.g. (100, 100, 100)
>
> True. It's more of a tradition. I am not sure if we really need to
> document each time the default value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/587/files#r41699339>.
>